Sources/TensorFlow/LazyTensor/LazyTensorOperation.swift

```diff
@@ -0,0 +1,551 @@
+        attrs[name] = Attribute.IntArray(value.map { Int($0) })
+    }
+    func updateAttribute(_ name: String, _ value: [Float]) {
+        attrs[name] = Attribute.FloatArray(value)
+    }
+    func updateAttribute(_ name: String, _ value: [Double]) {
+        attrs[name] = Attribute.DoubleArray(value)
+    }
+    func updateAttribute(_ name: String, _ value: [String]) {
+        attrs[name] = Attribute.StringArray(value)
+    }
+}
+
+extension LazyTensorOperation : TFTensorOperation {
+    private func lazyTensorHandle(_ input: _AnyTensorHandle) -> LazyTensor {
+        if let lazyHandle = input as? LazyTensor {
+            if case let LazyTensor.Handle.symbolic(
+                op, index, true) = lazyHandle.handle {
+                // We turn off liveness for the constructed LazyTensor,
+                // because it is only referenced internally as a part
+                // of the LazyTensorOperation input.
+                return LazyTensor(_lazy: op, index: index)
+            } else {
+                return lazyHandle
+            }
+        } else {
+            return LazyTensor(input._tfeTensorHandle)
+        }
+    }
+
+    func addInput(_ input: _AnyTensorHandle) {
+        addInput(lazyTensorHandle(input))
+    }
+
+    func addInput<Scalar: TensorFlowScalar>(_ input: Tensor<Scalar>) {
+        addInput(input.handle.handle)
+    }
+
+    func addInput(_ input: StringTensor) {
+        addInput(input.handle.handle)
+    }
+
+    func addInput(_ input: VariantHandle) {
+        addInput(input.handle)
+    }
+
+    func addInput(_ input: ResourceHandle) {
+        addInput(input.handle)
+    }
+
+    func addInputList<T: TensorArrayProtocol>(_ input: T) {
+        let lazyHandles = input._tensorHandles.map { lazyTensorHandle($0) }
+        inputs.append(Input.list(lazyHandles))
+    }
+
+    func updateAttribute(_ name: String, _ value: TensorDataType) {
+        attrs[name] = Attribute.TensorDataTypeValue(value)
+    }
+    func updateAttribute(_ name: String, _ value: TensorShape) {
+        assert(false, "Unimplemented TensorShape attribute.")
+    }
+    func updateAttribute(_ name: String, _ value: TensorShape?) {
+        assert(false, "Unimplemented TensorShape? attribute.")
+    }
+    func updateAttribute(_ name: String, _ value: [TensorDataType]) {
+        assert(false, "Unimplemented [TensorDataType] attribute.")
+    }
+    func updateAttribute(_ name: String, _ value: [TensorShape]) {
+        assert(false, "Unimplemented [TensorShape] attribute.")
+    }
+    func updateAttribute(_ name: String, _ value: [TensorShape?]) {
+        assert(false, "Unimplemented [TensorShape?] attribute.")
+    }
+    func updateAttribute<In: TensorGroup, Out: TensorGroup>(
+        _ name: String, _ value: (In) -> Out) {
+        // TODO:
+        assert(false, "Unimplemented [TFFunction] attribute.")
+    }
+
+    func execute() {}
+
+    func execute<T0 : TensorArrayProtocol>(
```

Comment:
```suggestion
    func execute<T0: TensorArrayProtocol>(
```
Same for other conformance constraints.